src/components/Checkbox/Checkbox.tsx

```diff
@@ -11,7 +11,7 @@ import {Error, Key, CheckboxHandles} from '../../types';
 
 import styles from './Checkbox.scss';
 
-export interface BaseProps {
+export interface CheckboxProps {
```

Comment:
This crops up in a few files - a BaseProps interface and the component's props that extend from that without adding anything news - so remove BaseProps.